### PR TITLE
Refine explanations and improve formatting in README.md

### PR DESCRIPTION
**Pull request by derskythe**
Hi, Thanks for the library.
For my part, I would like to offer updated documentation.
In my opinion, in this form it is much easier to read and easier to navigate.

What is done:
- [x] Added missing code blocks
- [x] Changing the format of code examples is was neccessary
- [x] Highlighting variable names and macro names in text
- [x] Refining links to section
- [x] Adjusting the indenting of API documentation listings for better readability

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5886,1100 +5947,1186 @@ A few macros are using recursion to work.
...
+* Throw an exception. In this case, the state of the object is unchanged
+* Set a global error variable and return
 
-NOTE: The last two cases are not properly fully supported yet.
-Throwing an exception is not fully supported yet 
-(Needs M\*LIB support to clear the skipped objects).
+> [!NOTE] 
+> The last two cases are not properly fully supported yet.
+> Throwing an exception is not fully supported yet
+> (Needs M\*LIB support to clear the skipped objects).
```
Comment (P-p-H-d) [inline README.md:7100]: The note is obsolete. Throwing an exception is now supported

The 3 last sentences should be : "The last case is not properly fully supported yet."

File: README.md [hunk 2]
```diff
@@ -8468,107 +8551,113 @@ To register an organization the *integration user* shall define a macro like thi
...
+```
 
 The following macro is defined and enabled to iterate over a container (See example):
 
-* each(item, container)
+* `each(item, container)`
 
-Note that only short names are defined in this header which may be incompatible with your own namespace (This header is still a WIP)
+> [!Note] 
+> that only short names are defined in this header which may be
```
Comment (P-p-H-d) [inline README.md:8657]: Remove "that"

**Review by P-p-H-d — CHANGES_REQUESTED** [review]
Thanks a lot for your huge rework on the documentation!

I tried to perform a review of it and find (few) issues in it I have listed below.
By reading the patch I also found some errors in the original documentation :)

File: README.md [hunk 3]
```diff
@@ -2927,139 +2959,141 @@ The current implementation is [RED-BLACK TREE](https://en.wikipedia.org/wiki/Red
...
 
-##### bool name\_it\_while\_p(const name\_it\_t it, const type data)
+##### `bool name_it_while_p(const name_it_t it, const type data)`
 
-Return true if 'it' references an element that is lower or equal than 'data'.
+Return true if `it` references an element that is lower or equal than `data`.
 Otherwise (or if it references no longer a valid element) it returns false.
 
-
+_________________
```
[the remark this answers is not in the document]
Comment (derskythe) [reply README.md:3101]: This line is used to further visually separate the chapters

image

File: README.md [hunk 4]
```diff
@@ -1895,10 +1900,10 @@ Example:
 #include <gmp.h>
 #include "m-list.h"
 
-#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)
-LIST_DUAL_PUSH_DEF(list_mpz, mpz_t,                                           \
-          (INIT(mpz_init), INIT_SET(mpz_init_set), SET(mpz_set), 
-              CLEAR(mpz_clear), OUT_STR(MPZ_OUT_STR)))
+#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)   \
+    LIST_DUAL_PUSH_DEF(list_mpz, mpz_t,                    \
```
Comment (P-p-H-d) [inline README.md:1904]: Same remark

File: README.md [hunk 5]
```diff
@@ -2053,10 +2058,10 @@ Example:
 #include <gmp.h>
 #include "m-array.h"
 
-#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)
-ARRAY_DEF(array_mpz, mpz_t,                                           \
-          (INIT(mpz_init), INIT_SET(mpz_init_set), SET(mpz_set), 
-              CLEAR(mpz_clear), OUT_STR(MPZ_OUT_STR)))
+#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)   \
+    ARRAY_DEF(array_mpz, mpz_t,                            \
```
Comment (P-p-H-d) [inline README.md:2062]: Same remark

File: README.md [hunk 6]
```diff
@@ -2711,10 +2716,10 @@ Example:
 #include "m-string.h"
 #include "m-tuple.h"
 
-#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)
-TUPLE_DEF2(pair,
-           (key_field, string_t, STRING_OPLIST),
-           (value_field, mpz_t, M_OPEXTEND(M_CLASSIC_OPLIST(mpz), OUT_STR(MPZ_OUT_STR))))
+#define MPZ_OUT_STR(stream, x) mpz_out_str(stream, 0, x)       \
+        TUPLE_DEF2(pair,(key_field, string_t, STRING_OPLIST),  \
```
Comment (P-p-H-d) [inline README.md:2720]: Same remark

**Pull request merged**